pkg/controller/bundlec/controller.go

```diff
@@ -110,6 +148,37 @@ func (c *Controller) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// lookupBundleByDeploymentByIndex returns a function that can be used to perform lookups of Bundles that contain
+// Deployment objects that reference ConfigMap/Secret objects.
+func (c *Controller) lookupBundleByDeploymentByIndex(byIndex byIndexFunc, indexName string, indexKey indexKeyFunc) func(runtime.Object) ([]runtime.Object, error) {
```

Comment:
This could probably be a nice util function in ctrl because we do something very similar in basically everywhere we have a LookupHandler

Reply:
I have checked the places where `LookupHandler` is used, but all of them are not like this one. They are simpler. We have duplication there and we should DRY it, but the logic is not the same as here. Or I'm missing something?